File: app/src/main/java/com/example/babajidemustapha/survey/features/Reports/adapters/ResponseAdapter.java
package com.example.babajidemustapha.survey.features.Reports.adapters;

import android.text.format.DateFormat;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.babajidemustapha.survey.R;
import com.example.babajidemustapha.survey.shared.room.entities.ResponseHeader;

import java.util.List;

public class ResponseAdapter extends RecyclerView.Adapter<ResponseAdapter.ViewHolder> {
    OnResponseSelectedListener mListener;
    private List<ResponseHeader> source;

    public ResponseAdapter(List<ResponseHeader> source, OnResponseSelectedListener mListener) {
        this.source = source;
        this.mListener = mListener;
    }

    public void add(ResponseHeader responseHeader) {
        source.add(responseHeader);
    }

    @NonNull
    @Override
    public ResponseAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ResponseAdapter.ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.response_list_item_layout, parent, false));
    }

    @Override
    public void onBindViewHolder(ResponseAdapter.ViewHolder holder, int position) {
        Log.e("id", source.get(position).getResponseId() + "");
        holder.name.setText("Response by: " + (source.get(position).getRespondentName() != null && !source.get(position).getRespondentName().isEmpty() ? source.get(position).getRespondentName() : "Anonymous"));
        holder.date.setText(DateFormat.format("dd/MM/yy", source.get(position).getDate()));
        holder.time.setText(DateFormat.format("HH:mm", source.get(position).getDate()));
    }

    @Override
    public int getItemCount() {
        return source.size();
    }

    public interface OnResponseSelectedListener {
        void onSelected(ResponseHeader responseHeader);
    }

    protected class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        TextView name;
        TextView date;
        TextView time;

        private ViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.response_name);
            date = itemView.findViewById(R.id.response_date);
            time = itemView.findViewById(R.id.response_time);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            mListener.onSelected(source.get(getAdapterPosition()));
        }
    }
}

File: app/src/main/java/com/example/babajidemustapha/survey/features/Reports/adapters/ResponseAdapter.kt
package com.example.babajidemustapha.survey.features.Reports.adapters

import android.text.format.DateFormat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.babajidemustapha.survey.R
import com.example.babajidemustapha.survey.shared.room.entities.ResponseHeader

class ResponseAdapter(private val source: MutableList<ResponseHeader>, var mListener: OnResponseSelectedListener) : RecyclerView.Adapter<ResponseAdapter.ViewHolder>() {
    fun add(responseHeader: ResponseHeader) {
        source.add(responseHeader)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.response_list_item_layout, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.name.text = "Response by: ${if (source[position].respondentName.isNullOrBlank().not()) source[position].respondentName else "Anonymous"}"
        holder.date.text = DateFormat.format("dd/MM/yy", source[position].date)
        holder.time.text = DateFormat.format("HH:mm", source[position].date)
    }

    override fun getItemCount(): Int {
        return source.size
    }

    interface OnResponseSelectedListener {
        fun onSelected(responseHeader: ResponseHeader?)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var name: TextView = itemView.findViewById(R.id.response_name)
        var date: TextView = itemView.findViewById(R.id.response_date)
        var time: TextView = itemView.findViewById(R.id.response_time)

        override fun onClick(view: View) {
            mListener.onSelected(source[adapterPosition])
        }

        init {
            itemView.setOnClickListener(this)
        }
    }
}